flagd/pkg/service/flag-sync/sync-multiplexer.go

```diff
@@ -201,6 +210,9 @@ func (r *Multiplexer) reFill() error {
 	for source, flags := range collector {
 		// store the corresponding metadata
 		metadata := r.store.GetMetadataForSource(source)
+		metadata["propagationTraceId"] = spanContext.TraceID()
+		metadata["propagationSpanId"] = spanContext.SpanID()
```

Comment:
I love this idea. I want to think very carefully about our naming though. I think `propagationTraceId` and `propagationSpanId` might be a bit generic. Maybe something more like `configurationChangedTraceId` and `configurationChangedSpanId` since our event name is `PROVIDER_CONFIGURATION_CHANGED`.

Reply:
I am not sure I agree, because the `PROVIDER_CONFIGURATION_CHANGED` is also a provider implementation detail. we also add this information on the first connection, which is a provider ready event.